include/oneapi/dpl/internal/dynamic_selection_impl/sycl_backend.h

```diff
@@ -35,25 +37,90 @@ class sycl_backend
     using execution_resource_t = resource_type;
     using resource_container_t = std::vector<execution_resource_t>;
 
+    using report_clock_type = std::chrono::steady_clock;
+    using report_duration = std::chrono::milliseconds;
+
+    static inline bool is_profiling_enabled = false;
   private:
-    class async_waiter
+    class async_waiter_base{
+        public:
+            virtual void wait() = 0;
```

Comment:
Why is `wait()` a virtual method? I do not see it used via the base class pointer; do I miss something?

Reply:
Fixed